python/mqt/debugger/dap/messages/launch_dap_message.py

```
@@ -77,7 +78,9 @@ def handle(self, server: DAPServer) -> dict[str, Any]:
             }
         if not self.stop_on_entry:
             server.simulation_state.run_simulation()
-        server.source_file = {"name": program_path.name, "path": self.program}
+        if not parsed_successfully:
```

Ruff (0.14.10)

Undefined name `parsed_successfully`

(F821)